pallets/maintenance-mode/src/lib.rs

```diff
@@ -71,6 +71,17 @@ pub mod pallet {
 	#[pallet::without_storage_info]
 	pub struct Pallet<T>(PhantomData<T>);
 
+	/// Pause and resume execution of XCM
+	pub trait PauseXcmExecution {
+		fn suspend_xcm_execution();
+		fn resume_xcm_execution();
```

Comment:
these extrinsics are fallible in cumulus_pallet_xcmp_queue but always succeed if they are called by the ControllerOrigin (which we set to root in our runtimes)

Reply:
We can make the trait output the failure (DispatchResult), and then inside palllet_maintenance_mode we just map it to an error from this pallet (Error::XCMPausingFailed). I feel this is more of a proper way of handling this, what do you think?

Reply:
I think we should just deposit an event. Then we also don't need to add a new error message, we can just emit the event if it is_err().